File: python/src/repository/student_repository.py
"""
Student Repository - manages student data in memory
"""
from typing import Dict, List, Optional
from ..model.student import Student


class StudentRepository:
    def __init__(self):
        # In-memory storage using dictionary
        self.students: Dict[str, Student] = {}
    
    def save(self, student: Student) -> Student:
        self.students[student.id] = student
        return student
    
    def find_by_id(self, id: str) -> Optional[Student]:
        return self.students.get(id)
    
    def find_all(self) -> List[Student]:
        return list(self.students.values())

File: javascript/src/repository/StudentRepository.js
/**
 * Student Repository - manages student data in memory
 */
class StudentRepository {
    constructor() {
        // In-memory storage using object
        this.students = {};
    }

    save(student) {
        this.students[student.id] = student;
        return student;
    }

    findById(id) {
        return this.students[id] || null;
    }

    findAll() {
        return Object.values(this.students);
    }
}

module.exports = StudentRepository;
